lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -234,6 +227,22 @@ def submit(actions)
       actions_to_retry
     end
 
+    def report_doc_level_error(message, action, status, response)
+      # To support bwc, we check if DLQ exists. otherwise we log and drop event (previous behavior)
+      if @dlq_writer
+        # TODO: Change this to send a map with { :status => status, :action => action } in the future
+        @dlq_writer.write(action[2], "#{message} status: #{status}, action: #{action}, response: #{response}")
```

Comment:
I'm not a huge fan of this "report_doc_level_error" to be writing data to the DLQ. Maybe we can have this method only take care of the `else` case in this if, where the DLQ isn't  available.
Another alternative is to call this method something like `handle_doc_level_error`. the word "handle" works for everything, yay

Reply:
Yes, `handle_` is a better name

Reply:
Actually this method is specifically for statuses that are candidates for the DLQ. I'll go even further and rename it to `handle_dlq_status`. It logs if DLQ is not enabled, and it sends to DLQ if it's enabled.